src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1827,8 +1829,10 @@ Task<ReadOnlyCollection<NpgsqlDbColumn>> GetColumnSchema(bool async, Cancellatio
 #endif
         => GetSchemaTable(async: true, cancellationToken);
 
-    [UnconditionalSuppressMessage(
-        "Composite type mapping currently isn't trimming-safe, and warnings are generated at the MapComposite level.", "IL2026")]
+#if NET8_0_OR_GREATER
+    [DynamicDependency(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.PublicProperties, typeof(Type))]
```

Comment:
What's this for? Maybe remove the #if?

Reply:
We can't use the #if, it's an 8.0 attribute

Reply:
[DynamicDependency]? [Are you sure](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicdependencyattribute?view=net-7.0), what am I missing?

Reply:
Yep I've lost it 😂